internal/xds/translator/metadata.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+const (
+	envoyGatewayXdsMetadataNamespace      = "envoy-gateway"
+	envoyGatewayXdsMetadataKeyKind        = "kind"
+	envoyGatewayXdsMetadataKeyName        = "name"
+	envoyGatewayXdsMetadataKeyNamespace   = "namespace"
+	envoyGatewayXdsMetadataKeyAnnotations = "annotations"
+	envoyGatewayXdsMetadataKeySectionName = "sectionName"
+	envoyGatewayXdsMetadataKeyRoute       = "route"
```

Comment:
why are these keys needed ?
as a consumer I may want to access the `envoy-gateway` metadata, but I may not know if the xds resource maps to a `gateway` or `route` or both